### DISCUSSION TITLE
Examples with adapters

### DISCUSSION THREAD

**Discussion by domcir** [bytedeco/javacpp#587] [Q&A] [answered]
Hi

I'm looking for examples with adapters when a C++ type is maped to a java native type like java.time.LocalDate.

I would like to map my class MyDate when used as a parameter in for examle:
void useMyDate(MyDate date);
it would be converted in Java to:
void usemyDate(LocalDate date)

I'm not sure how to add it to JavaCpp?

For example:
```
class MyDate{
    public:
        int _day{0};
        int _month{0};
        int _year{0};
        MyDate(
            int day,
            int month,
            int year
        ) : 
            _day(day),
            _month(month),
            _year(year)
        { }

};
```
Thanks
Dominik

**Comment by saudet** [comment, marked as answer]
JavaCPP adapters cannot use arbitrary Java classes like that. The way we usually go about this is by creating a higher-level API on top of the low-level one that JavaCPP generates. Other tools such as JavaPoet may help you with that: https://github.com/square/javapoet

**Reply by domcir** [reply]
Thanks, it's very useful.
